### DISCUSSION TITLE
PyScript & MicroPython

### DISCUSSION THREAD

**Discussion by jimmo** [micropython/micropython#9909] [Announcements and news]
Exciting announcement from Anaconda today about using MicroPython as a backed for PyScript, which allows you to write Python in the browser.

https://www.anaconda.com/blog/pyscript-updates-bytecode-alliance-pyodide-and-micropython

See the demo at https://pyscript.net/tech-preview/micropython/ which also includes an example of running ulab (@v923z).

**Comment by sosi-deadeye** [comment]
Cool, in one of the examples they even use ulab.numpy, which is compiled into the wasm.

**Comment by cwalther** [comment]
Nice, looks like they have solved some of the same problems I was working on and some things have already made it upstream. I wonder if they are going to have asynchronous execution, which is what I was stuck on.

**Reply by Neon22** [reply]
async is now in the top level loop  (2024.8.2) - new projects made at [pyscript.com](https://pyscript.com/) use 2024.8.2 as default version so you can experiment painlessly there.
